src/pygama/flow/data_loader.py

```diff
@@ -0,0 +1,1060 @@
+        some kind of query to retrieve parameters of interest for our file list,
+        and return some tables.
+        """
+        pass
+
+    def load_cal_pars(self, query):  # TODO
+        """
+        access the cal_pars parameter database, run a query, and return some tables.
+        """
+        pass
+
+    def skim_waveforms(self, mode: str = "hit", hit_list=None, evt_list=None):  # TODO
+        """
+        handle this one separately because waveforms can easily fill up memory.
+        """
+        if mode == "hit":
+            pass
+        elif mode == "evt":
+            pass
+        pass
+
+    def browse(self, query, dsp_config=None):  # TODO
+        """
+        Interface between DataLoader and WaveformBrowser.
+        """
+        pass
+
+    def reset(self):
+        self.file_list = None
+        self.table_list = None
+        self.cuts = None
+        self.merge_files = False
+        self.output_format = "lgdo.Table"
+        self.output_columns = None
+
+
+if __name__ == "__main__":
+    doc = """
+    Demonstrate usage of the `DataLoader` class.
+    This could be what we initially run at LNGS - it would try to do the `os.walk`
+    method over the existing files, and e.g. scan for existence of various files
+    in different stages.  More advanced tests would be moved to a notebook or
+    separate script.
+    """
+
+    dl = DataLoader(
+        config="../../../../loader_config.json",
+        fileDB_config="fileDB_cfg.json",
+        fileDB="fileDB.lh5",
+    )
+    dl.show_filedb()
+
+    # dl = DataLoader(config="../../../../loader_config.json",
+    #                 fileDB_config="../../../../fileDB_config_copy.json")
+    # dl.show_fileDB()
+
+    # pd.set_option('display.max_colwidth', 10)
+    # dl.fileDB.get_tables_columns()
+    # dl.show_fileDB()
+
+    # dl.fileDB.to_disk("fileDB_cfg.json", "fileDB.lh5")
+    # dl2 = DataLoader(config="../../../../loader_config.json", fileDB_config="fileDB_cfg.json", fileDB="fileDB.lh5")
+    # dl2.show_fileDB()
+
+    dl.set_files("file_status == 26 and timestamp == '20220716T130443Z'")
+    dl.show_file_list(
+        [
+            "period",
+            "run",
+            "type",
+            "raw_tables",
+            "dsp_tables",
+            "tcm_tables",
+            "file_status",
+        ]
+    )
+
+    dl.set_datastreams(np.arange(45), "ch")
+
+    # hit_el = dl.gen_entry_list()
+
+    dl.set_cuts({"hit": "trapEmax > 1000"})
+
+    # hit_cut_el = dl.gen_entry_list()
+
+    # tcm_el = dl.gen_entry_list(tcm_level="tcm")
+
+    cols = ["daqenergy", "trapEmax", "channel", "waveform"]
+    dl.set_output(fmt="lgdo.Table", merge_files=False, columns=cols)
+    # dl.set_cuts({"hit": "daqenergy > 100"})
+
+    # hit_cut_el = dl.gen_entry_list(save_output_columns=True)
+
+    tcm_el = dl.gen_entry_list(tcm_level="tcm", save_output_columns=True)
+
+    lout = dl.load(tcm_el, in_mem=True, rows="hit", tcm_level="tcm")
```

Comment:
We'll remove this part later. It would be really nice if you could refactor this tiny tutorial as an "examples" section in the `DataLoader` class docstring:
```
class DataLoader:
    """One-line description.
    
    Longer description.
    
    Examples
    --------
    >>> from pygama.flow import DataLoader
    >>> dl = DataLoader()
    >>> ...
    ```